File: python/ex-2.py

"""
Exercise - 2

2- Search and count the word available in file. 
"""

# read file content
file = open("words.txt", "rt")
data = file.read()

# split file data to list containing words
words = data.split()

print('total number of words in text file :', len(words))

# enter word to search
searchWord = input("input word to search: ")

# check if word is present in list
if searchWord in words:
    # if word is present then print word and total number of occurance
    print('Word found: ',searchWord," occurances: ",words.count(searchWord));
else:
    print('Word not found')

File: go/ex-2.go
package main

import (
	"fmt"
	"os"
	"strings"
)

/**
Exercise - 2

2- Search and count the word available in file.
**/

func check(e error) {
	if e != nil {
		panic(e)
	}
}

func main() {
	dat, err := os.ReadFile("words.txt")
	check(err)
	content := string(dat)

	println("Enter word to replace: ")
	var findWord string
	fmt.Scanln(&findWord)

	println("Enter replacement word: ")
	var replacementWord string
	fmt.Scanln(&replacementWord)
	output := ""
	words := strings.Split(content, " ")
	for i, word := range words {
		if strings.Contains(word, findWord) {
			words[i] = strings.Replace(word, findWord, replacementWord, 1)
		} else {
			output += word
		}
	}
	for i := 1; i < len(words); i++ {
		println("words[i] " + words[i])
		output += " " + words[i]
	}
	os.WriteFile("words-out", []byte(output), 0644)
}
